geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisString.java

```diff
@@ -469,7 +469,7 @@ public void handleSetExpiration(SetOptions options) {
 
   ////// methods that modify the "value" field ////////////
 
-  protected void valueAppend(byte[] bytes) {
+  protected synchronized void valueAppend(byte[] bytes) {
```

Comment:
My understanding is that this method does not need to be synchronized because it does not modify "value" in place. It just reads value and copies it into a new byte array and then at the end sets the value field. We do this same thing in other places (for example incr). Since our concern here is thread safety with "toData", which only reads "value", we would only cause corruption for toData if we modified "value" in place. So this method does not need to be synchronized. But this line of applyReplaceByteArrayAtOffsetDelta does need to be sync'd:
      System.arraycopy(valueToAdd, 0, value, offset, valueToAdd.length);

Reply:
Ah, you're right. I'll remove the synchronized from this method and add a synchronized block around that line in `applyReplaceByteArrayAtOffsetDelta()`.